packages/swapper/src/swappers/PortalsSwapper/getPortalsTradeQuote/getPortalsTradeQuote.ts

```diff
@@ -86,124 +91,171 @@ export async function getPortalsTradeQuote(
-
-    const tradeQuote: TradeQuote = {
-      id: orderId,
-      receiveAddress: input.receiveAddress,
-      affiliateBps,
-      potentialAffiliateBps,
-      rate: inputOutputRate,
-      slippageTolerancePercentageDecimal,
-      steps: [
-        {
-          accountNumber,
-          allowanceContract,
-          rate: inputOutputRate,
-          buyAsset,
-          sellAsset,
-          buyAmountBeforeFeesCryptoBaseUnit,
-          buyAmountAfterFeesCryptoBaseUnit,
-          sellAmountIncludingProtocolFeesCryptoBaseUnit:
-            input.sellAmountIncludingProtocolFeesCryptoBaseUnit,
-          feeData: {
-            networkFeeCryptoBaseUnit,
-            protocolFees: {
-              [protocolFeeAsset.assetId]: {
-                amountCryptoBaseUnit: feeAmount,
-                asset: protocolFeeAsset,
-                requiresBalance: false,
-              },
+  const adapter = assertGetEvmChainAdapter(chainId)
+  const { average } = await adapter.getGasFeeData()
+
+  const networkFeeCryptoBaseUnit = evm.calcNetworkFeeCryptoBaseUnit({
+    ...average,
+    supportsEIP1559: Boolean(supportsEIP1559),
+    // times 1 isn't a mistake, it's just so we can write this comment above to mention that Portals already add a
+    // buffer of ~15% to the gas limit
+    gasLimit: bnOrZero(gasLimit).times(1).toFixed(),
+  })
+
+  const slippageTolerancePercentageDecimal = bnOrZero(slippageTolerancePercentage)
+    .div(100)
+    .toString()
+
+  const tradeQuote: TradeQuote = {
+    id: orderId,
+    receiveAddress: input.receiveAddress,
+    affiliateBps,
+    potentialAffiliateBps,
+    rate: inputOutputRate,
+    slippageTolerancePercentageDecimal,
+    steps: [
+      {
+        accountNumber,
+        allowanceContract,
+        rate: inputOutputRate,
+        buyAsset,
+        sellAsset,
+        buyAmountBeforeFeesCryptoBaseUnit,
+        buyAmountAfterFeesCryptoBaseUnit,
+        sellAmountIncludingProtocolFeesCryptoBaseUnit:
+          input.sellAmountIncludingProtocolFeesCryptoBaseUnit,
+        feeData: {
+          networkFeeCryptoBaseUnit,
+          protocolFees: {
+            [protocolFeeAsset.assetId]: {
+              amountCryptoBaseUnit: feeAmount,
+              asset: protocolFeeAsset,
+              requiresBalance: false,
             },
           },
-          source: SwapperName.Portals,
-          estimatedExecutionTimeMs: undefined, // Portals doesn't provide this info
-          portalsTransactionMetadata: tx,
         },
-      ] as SingleHopTradeQuoteSteps,
-    }
-
-    return Ok(tradeQuote)
-  } catch (err) {
-    return Err(
-      makeSwapErrorRight({
-        message: 'failed to get Portals quote',
-        cause: err,
-        code: TradeQuoteError.NetworkFeeEstimationFailed,
-      }),
-    )
+        source: SwapperName.Portals,
+        estimatedExecutionTimeMs: undefined, // Portals doesn't provide this info
+        portalsTransactionMetadata: tx,
+      },
+    ] as SingleHopTradeQuoteSteps,
   }
```

Comment:
There should be no other parts that can throw exceptions, and the one that can is now explicitly handled with .catch() chaining